research-hub-web/src/app/services/cer-graphql.service.ts

```diff
@@ -31,10 +32,17 @@ export class CerGraphqlService {
    * Dynamically pushes the SubHubs and the SubHub child pages to the application's routing array
    */
   public async pushSubHubRoutes(): Promise<void> {
+    this.hasPushedSubhubRoutes = true;
     const routes = this.router.config;
     await this._generateSubHubMapAndRoutes(); // Generate _subHubMap.map and _subHubMap.routes
-    this._subHubMap.routes.forEach(route => { routes.push(route); }); // Push the new routes to the application's routes.
-    this.router.resetConfig(routes);
+    const wildcardRouteIdx = routes.findIndex((route) => route.path === "**");
+    let  newRoutes;
+    if (wildcardRouteIdx > -1) {
+      newRoutes = routes.slice(0, wildcardRouteIdx).concat(this._subHubMap.routes, routes.slice(wildcardRouteIdx, routes.length))
```

Comment:
😕 😨

Reply:
Why is Sam sad about this.. should we be worried?

Reply:
I added comments. :sweat_smile: